packages/components/addon/components/hds/code-block/index.hbs

```diff
@@ -0,0 +1,30 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<div class={{this.classNames}} ...attributes>
+  <div class="hds-code-block__header">
+    {{~yield (hash Title=(component "hds/code-block/title"))~}}
+    {{~yield (hash Description=(component "hds/code-block/description"))~}}
+  </div>
+
+  <pre
+    class="hds-code-block__code"
+    {{style maxHeight=@maxHeight}}
+    contentEditable={{not this.isReadOnly}}
+    data-line={{@highlightLines}}
+    id={{if @hasCopyButton this.preCodeId null}}
+    tabindex="0"
+  >
+    {{~! ~}}
+    <code {{did-insert this.setPrismCode}} {{did-update this.setPrismCode this.code @language}}>
+      {{~this.prismCode~}}
+    </code>
+    {{~! ~}}
+  </pre>
+
+  {{#if @hasCopyButton}}
+    <Hds::CodeBlock::CopyButton @textToCopy={{@value}} aria-describedby={{this.preCodeId}} />
```

Comment:
Should this be `this.code` which is whitespace-normalized, or `@value` which is the one passed as argument? 

```suggestion
    <Hds::CodeBlock::CopyButton @textToCopy={{this.code}} aria-describedby={{this.preCodeId}} />
```

Reply:
See also comment about having only `@textToCopy` when the content is editable

Reply:
Refactored to use a target in a09ac93e4d7ee084f823b6cd7ec48b0885a74d19